v3/src/components/data-display/models/data-configuration-model.ts

```diff
@@ -357,7 +357,9 @@ export const DataConfigurationModel = types
               (extraPrimaryValue === "__main__" ||
                 dataset?.getStrValue(aCaseData.caseID, extraPrimaryAttrID) === extraPrimaryValue) &&
               (extraSecondaryValue === "__main__" ||
-                dataset?.getStrValue(aCaseData.caseID, extraSecondaryAttrID) === extraSecondaryValue)
+                dataset?.getStrValue(aCaseData.caseID, extraSecondaryAttrID) === extraSecondaryValue) &&
+              (!legendCat || legendCat === "" ||
+                dataset?.getStrValue(aCaseData.caseID, self.attributeID("legend")) === legendCat)
```

Comment:
```suggestion
              (!legendCat || dataset?.getStrValue(aCaseData.caseID, self.attributeID("legend")) === legendCat)
```
Explicit test for empty string is redundant.